Offset piezo / focus lock performance improvements

barentine opened the pull request:
Addresses issue #313, maybe.

**Is this a bugfix or an enhancement?**
enhancement for the sake of bugfix

**Proposed changes:**
- use `request.Session.get` instead of `request.get` for socket keep-alives in `RLPIDFocusLockClient` 
- use `request.Session.get` instead of `request.get` for socket keep-alives in `OffsetPiezoClient` 
- connect focus lock to `frameWrangler.onFrameGroup` rather than `onFrame`
- make `frameWrangler` polling interval an attribute so we can change it if we like
- change `frameWrangler._polling_interval` in init_htsms_focus_lock
- move frameWrangler signal doc into docstring and clarify the part that I misunderstood earlier
- add minimal (typing) docstrings to `microscope` and `GUIInitTask` to aid linter/IDE shortcuts






**Checklist:**

- [ ] Tested with numpy=1.14
- [ ] Tested on python 2.7 and 3.6
- [ ] Tested with wx=3.x and wx=4.x [if UI code]
- [ ] Does the PR avoid variable renaming in existing code, whitespace changes, and other forms of tidying? [There is a place for code tidying, but it makes reviewing 
much simpler if this is kept separate from functional changes]

If an enhancement (or non-trivial bugfix):

- [x] Has this been discussed in advance (feature request, PR proposal, email, or direct conversation)?
- [ ] Does this change how users interact with the software? How will these changes be communicated?
- [x] Does this maintain backwards compatibility with old data?
- [ ] Does this change the required dependencies?
- [ ] Are there any other side effects of the change?

barentine committed:
use a request session to have sockets kept-alive for the focus lock client

barentine committed:
change poll interval to frameWrangler attribute and add some doc

barentine committed:
connect focus lock to onFrameGroup

barentine committed:
microscope.frameWrangler doc / test everyone's linters re: init gui

barentine committed:
see if we can't run the focus lock a bit faster

barentine committed:
use requests session for get keep-alives in OffsetPiezoClient

barentine committed:
changelog

barentine commented:
converted to draft while I re-tune the htsms focus lock - performance is nice, might even be able to add in the fastGraph stuff again

barentine committed:
update focus lock servo tuning, mix back in the fastgraph now that our performance is good enough it doesn't break the bank

barentine commented:
things are running very nicely, though the D was annoying enough I left it at the ZN PI servo for now. Should be all set @David-Baddeley

David-Baddeley commented on PYME/Acquire/frameWrangler.py:
```diff
@@ -54,6 +54,31 @@
+        Called on each new frame group (once per polling interval) - use for 
+        updateing GUIs etc.
+    onStop : dispatch.Signal
+    _polling_interval : float
```
It's a bit odd having a private attribute with no way of setting documented (leading underscore implies that it should not be accessed from outside the class)

David-Baddeley replied:
This is also a bit of a 'here be dragons' type thing - I wouldn't want to encourage people to change this if it were to be made a public attribute as the results might not be super predictable.

David-Baddeley replied:
Best approach might be to leave as an undocumented (or documented with a comment but not in docstring) private attribute.

David-Baddeley commented:
Should also really get this updated and into core docs (graphically illustrates frame timing)
[PYMEAcquireSchem.pdf](https://github.com/python-microscopy/python-microscopy/files/4910444/PYMEAcquireSchem.pdf)

David-Baddeley committed:
Update frameWrangler.py

David-Baddeley committed:
Removed changelog entry

changes won't effect how anyone uses the software, probably fall under the too small to care about category re changelog.
